### ISSUE TITLE
F18 (now Flang) upstreamed

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/flang-feedstock#16]
FWIW Flang entered LLVM mainline ( https://github.com/llvm/llvm-project/tree/main/flang/ ). Guessing it will be available in LLVM 11.0.

**Comment by isuruf** [comment]
Not useful at the moment as it is only a parser at the moment and code generation is a wip.

**Comment by jakirkham** [comment]
Sounds like this is [now available in LLVM 11.0]( https://www.phoronix.com/scan.php?page=news_item&px=LLVM-11.0-Released ).

**Comment by h-vetinari** [comment]
The very first bits of code generation made it for LLVM 14, but realistically nothing usable before LLVM 15.

**Comment by jakirkham** [comment]
There were a couple talks at a recent [LLVM Developer Meeting]( https://llvm.org/devmtg/2022-05/ ). Not exactly sure when the slides/videos will be available. Maybe in [a couple weeks?]( https://discourse.llvm.org/t/announcing-the-2022-eurollvm-program/61741/4 )

In particular this one:

> How to write a new compiler driver? The LLVM Flang perspective.
> Andrzej Warzynski, Arm
>
> When LLVM Flang (then "F18") was merged into LLVM as a sub-project in 2020, it had no compiler driver. Two years later, Flang enjoys a driver that integrates all components of the Flang sub-project, can generate executables and that shares the driver logic with Clang. In this presentation I will walk you through our journey.

Also this one:

> Flang update
> Kiran Chandramohan, Arm
>
> F18/Flang was accepted as the Fortran frontend of LLVM in 2019. The project joined the monorepo in 2020. Development of a large chunk of code dealing with the lowering of the parse-tree to FIR (Fortran IR) continued in a fork of the llvm-project. In the last year, a significant effort went in to upstreaming this fork. Most of this code is upstreamed and all development has shifted to the llvm-project monorepo. Much development effort has also gone into adding a runtime and lowering for Fortran 95. In parallel, great progress has been achieved on adding a driver, support for OpenMP 1.1+ and OpenACC. In this talk, I will summarize the developmental activities for the last year, the current status and future work.

**Comment by h-vetinari** [comment]
The first rc of LLVM 15 just dropped. I'll probably need a couple bits and pieces in place first, but I'd like to build the rc's of LLVM flang on this feedstock. Could someone create an `rc` branch for me to target, please? Also happy to join as maintainer and do it myself, if that's preferred.

**Comment by jakirkham** [comment]
Given LLVM 15 has since come out, maybe this is easier to explore

**Comment by h-vetinari** [comment]
We've got first builds of flang 17 as of #28 now.

**Comment by h-vetinari** [comment]
Closing this one as completed
